c_pal_ll/linux/devdoc/sync_linux_requirements.md

```diff
@@ -30,6 +33,24 @@ MOCKABLE_FUNCTION(, WAIT_ON_ADDRESS_RESULT, wait_on_address, volatile_atomic int
 
 **SRS_SYNC_LINUX_43_004: [** Otherwise, `wait_on_address` shall return `WAIT_ON_ADDRESS_ERROR`.**]**
 
+## wait_on_address_64
+
+```c
+MOCKABLE_FUNCTION(, WAIT_ON_ADDRESS_RESULT, wait_on_address_64, volatile_atomic int64_t*, address, int64_t, compare_value, uint32_t, timeout_ms)
+```
+
+**SRS_SYNC_LINUX_05_001: [** `wait_on_address_64` shall initialize a `timespec` struct with `.tv_nsec` equal to `timeout_ms* 10^6`. **]**
+
+**SRS_SYNC_LINUX_05_002: [** `wait_on_address_64` shall call `syscall` to wait on value at `address` to change than the one provided in `compare_value`. **]**
+
+**SRS_SYNC_LINUX_05_003: [** `wait_on_address_64` shall return true if the value at `address` changes than value at `compare_value`. **]**
+
+**SRS_SYNC_LINUX_05_004: [** If `syscall` returns a non-zero value and errno is `EAGAIN`, `wait_on_address_64` shall return `WAIT_ON_ADDRESS_OK`. **]**
+
+**SRS_SYNC_LINUX_05_005: [** If `syscall` returns a non-zero value and errno is `ETIMEDOUT`, `wait_on_address_64` shall return `WAIT_ON_ADDRESS_TIMEOUT`. **]**
+
+**SRS_SYNC_LINUX_05_006: [** Otherwise, `wait_on_address_64` shall return `WAIT_ON_ADDRESS_ERROR`. **]**
```

Comment:
These specs seem to be the same as the as the specs for `wait_on_address`. Is it possible to extract them into a common function so that we can avoid code duplication?

Reply:
same comment applies to other `_64` APIs

Reply:
Not possible since most functions are internally calling other _64 or _32 functions. Refer to interlocked.h or interlocked_win32.c which has complete implementation of _16, _32, _64